author_handlers_test.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"bytes"
+	"errors"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+var curatedAuthorsTransformer *httptest.Server
+
+var uuids = []string{martinWolf.Uuid, lucyKellaway.Uuid}
+var expectedStreamOutput = `{"id":"` + martinWolf.Uuid + `"} {"id":"` + lucyKellaway.Uuid + `"} `
+
+var martinWolfIdentifier = identifier{tmeAuthority, "Q0ItMDAwMDkwMA==-QXV0aG9ycw=="}
+
+var transformedMartinWolf = person{
+	Uuid:           "daf5fed2-013c-468d-85c4-aee779b8aa53",
+	Name:           "Martin Wolf",
+	EmailAddress:   "martin.wolf@ft.com",
+	TwitterHandle:  "@martinwolf_",
+	Description:    "Martin Wolf is chief economics commentator at the Financial Times, London.",
+	DescriptionXML: `<p>Martin Wolf is chief economics commentator at the Financial Times, London.</p>`,
+	ImageUrl:       "https://next-geebee.ft.com/image/v1/images/raw/fthead:martin-wolf?source=next",
+	Identifiers:    []identifier{martinWolfIdentifier},
+}
+
+type MockedBerthaService struct {
+	mock.Mock
+}
+
+func (m *MockedBerthaService) getAuthorsUuids() []string {
+	args := m.Called()
+	return args.Get(0).([]string)
+}
+
+func (m *MockedBerthaService) getAuthorByUuid(uuid string) author {
+	args := m.Called(uuid)
+	return args.Get(0).(author)
+}
+
+func (m *MockedBerthaService) getAuthorsCount() (int, error) {
+	args := m.Called()
+	return args.Int(0), args.Error(1)
+}
+
+func (m *MockedBerthaService) checkConnectivity() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+type MockedTransformer struct {
+	mock.Mock
+}
+
+func (m *MockedTransformer) authorToPerson(a author) (person, error) {
+	args := m.Called(a)
+	return args.Get(0).(person), args.Error(1)
+}
+
+func startCuratedAuthorsTransformer(bs *MockedBerthaService, mt *MockedTransformer) {
+	ah := authorHandler{
+		authorsService: bs,
+		transformer:    mt,
+	}
+	r := mux.NewRouter()
+	r.HandleFunc("/transformers/authors/__count", ah.getAuthorsCount).Methods("GET")
+	r.HandleFunc("/transformers/authors/__ids", ah.getAuthorsUuids).Methods("GET")
+	r.HandleFunc("/transformers/authors/{uuid}", ah.getAuthorByUuid).Methods("GET")
+	curatedAuthorsTransformer = httptest.NewServer(r)
```

Comment:
I don't think your tests should set up the routing again - you should be testing the routing used in the actual app. Otherwise, I can change the routing without breaking your tests. 

We have lots of examples of this (basically, encapsulate the routing setup in a function you can also call from your tests)